### PR TITLE
Corrected links in old Demos

### PR DESCRIPTION
**Pull request by ikurecic**
Fixing issues with links to in-page section anchors.

### CODE REVIEW

**Comment by Qottmann** [comment]
not sure whats going on [here](https://github.com/PennyLaneAI/qml/actions/runs/10140514148/job/28035958847?pr=1173), seems to be unrelated to this PR though

Traceback (most recent call last):
  File "/home/runner/work/qml/qml/venv/lib/python3.10/site-packages/sphinx/config.py", line 350, in eval_config_file
    exec(code, namespace)
  File "/home/runner/work/qml/qml/conf.py", line 21, in <module>
    from pennylane import PennyLaneDeprecationWarning
ModuleNotFoundError: No module named 'pennylane'

**Comment by ikurecic** [comment]
> not sure whats going on [here](https://github.com/PennyLaneAI/qml/actions/runs/10140514148/job/28035958847?pr=1173), seems to be unrelated to this PR though
> 
> Traceback (most recent call last): File "/home/runner/work/qml/qml/venv/lib/python3.10/site-packages/sphinx/config.py", line 350, in eval_config_file exec(code, namespace) File "/home/runner/work/qml/qml/conf.py", line 21, in from pennylane import PennyLaneDeprecationWarning ModuleNotFoundError: No module named 'pennylane'

Yikes. Yeah, it seems to be unrelated. I'll follow up. Thanks!

**Comment by DSGuala** [comment]
When I tested locally, keeping the two slashes allows the link to point to the demo, but one slash just doesn't show up as a link.

**Comment by DSGuala** [comment]
> For [cross-referencing sections](https://www.sphinx-doc.org/en/master/usage/referencing.html), I don't think this is allowed syntax with Sphinx 🤔 (but could be wrong). Instead, I typically use [`:ref:`](https://www.sphinx-doc.org/en/master/usage/referencing.html#role-ref)

Don't see any examples of `` :ref:`...#` `` to a section :thinking: 
and found some broken :ref: links.
- https://pennylane.ai/qml/demos/tutorial_quantum_chemistry/ ``:ref:`hamiltonian` ``-
- https://pennylane.ai/qml/demos/tutorial_quantum_natural_gradient/ ``:ref:`Concepts section <glossary_variational_circuit>` ``

**Comment by ikurecic** [comment]
I just made the links external, it seems like that works reliably at the moment.

**Review by ashishks0522 — APPROVED** [review]

File: demonstrations/tutorial_quantum_chemistry.py [hunk 1]
```diff
@@ -255,7 +255,7 @@
 ##############################################################################
 # In this case, since we have truncated the basis of molecular orbitals, the resulting
 # observable is an approximation of the Hamiltonian generated in the
-# section :ref:`hamiltonian`.
+# section `Building the Hamiltonian <https://qml-build-previews.pennylane.ai/pull_request_build_preview/1173/qml/demos/tutorial_quantum_chemistry/#building-the-hamiltonian>`__.
```
Comment (josh146) [inline demonstrations/tutorial_quantum_chemistry.py:258]: @ikurecic I assume this shouldn't be a PR preview link?

**Review by josh146 — APPROVED** [review]

**Pull request merged**